modules/fluid_properties/include/userobjects/CO2FluidProperties.h

```diff
@@ -309,49 +309,50 @@ class CO2FluidProperties : public SinglePhaseFluidPropertiesPT
+  const std::vector<Real> _theta0{3.15163, 6.11190, 6.77708, 11.32384, 27.08792};
 
   /// Coefficients for the residual component of the Helmholtz free energy
-  std::vector<Real> _n1{0.38856823203161,
-                        2.9385475942740,
-                        -5.5867188534934,
-                        -0.76753199592477,
-                        0.31729005580416,
-                        0.54803315897767,
-                        0.12279411220335};
-  std::vector<Real> _d1{1.0, 1.0, 1.0, 1.0, 2.0, 2.0, 3.0};
-  std::vector<Real> _t1{0.0, 0.75, 1.0, 2.0, 0.75, 2.0, 0.75};
-  std::vector<Real> _n2{
+  const std::vector<Real> _n1{0.38856823203161,
+                              2.9385475942740,
+                              -5.5867188534934,
+                              -0.76753199592477,
+                              0.31729005580416,
+                              0.54803315897767,
+                              0.12279411220335};
+  const std::vector<unsigned int> _d1{1, 1, 1, 1, 2, 2, 3};
+  const std::vector<Real> _t1{0.0, 0.75, 1.0, 2.0, 0.75, 2.0, 0.75};
+  const std::vector<Real> _n2{
       2.1658961543220,      1.5841735109724,    -0.23132705405503,   0.058116916431436,
       -0.55369137205382,    0.48946615909422,   -0.024275739843501,  0.062494790501678,
       -0.12175860225246,    -0.37055685270086,  -0.016775879700426,  -0.11960736637987,
       -0.045619362508778,   0.035612789270346,  -0.0074427727132052, -0.0017395704902432,
       -0.021810121289527,   0.024332166559236,  -0.037440133423463,  0.14338715756878,
       -0.13491969083286,    -0.023151225053480, 0.012363125492901,   0.0021058321972940,
       -0.00033958519026368, 0.0055993651771592, -0.00030335118055646};
-  std::vector<Real> _d2{1.0, 2.0, 4.0, 5.0, 5.0, 5.0, 6.0, 6.0, 6.0, 1.0, 1.0,  4.0, 4.0, 4.0,
-                        7.0, 8.0, 2.0, 3.0, 3.0, 5.0, 5.0, 6.0, 7.0, 8.0, 10.0, 4.0, 8.0};
-  std::vector<Real> _t2{1.5, 1.5, 2.5, 0.0,  1.5,  2.0,  0.0,  1.0,  2.0,  3.0, 6.0, 3.0,  6.0, 8.0,
-                        6.0, 0.0, 7.0, 12.0, 16.0, 22.0, 24.0, 16.0, 24.0, 8.0, 2.0, 28.0, 14.0};
-  std::vector<Real> _c2{1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 2.0, 2.0, 2.0, 2.0, 2.0,
-                        2.0, 2.0, 3.0, 3.0, 3.0, 4.0, 4.0, 4.0, 4.0, 4.0, 4.0, 5.0, 6.0};
-  std::vector<Real> _n3{
+  const std::vector<unsigned int> _d2{1, 2, 4, 5, 5, 5, 6, 6, 6, 1, 1,  4, 4, 4,
+                                      7, 8, 2, 3, 3, 5, 5, 6, 7, 8, 10, 4, 8};
+  const std::vector<Real> _t2{1.5,  1.5,  2.5,  0.0,  1.5,  2.0, 0.0, 1.0,  2.0,
+                              3.0,  6.0,  3.0,  6.0,  8.0,  6.0, 0.0, 7.0,  12.0,
+                              16.0, 22.0, 24.0, 16.0, 24.0, 8.0, 2.0, 28.0, 14.0};
+  const std::vector<unsigned int> _c2{1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2,
+                                      2, 2, 3, 3, 3, 4, 4, 4, 4, 4, 4, 5, 6};
+  const std::vector<Real> _n3{
```

Comment:
Not to over-optimize, but you might start using `std::array` for these const fixed sized arrays. Maybe the compiler is doing something smart, but I'm not sure. `std::array` is a new C++11 type that's for the special case where you have a statically sized array. The compiler can allocate that space on the stack as opposed to the heap (which is where normal `std::vectors` are allocated). This subtle difference can make a big difference in speed. Take a look when you get a chance.

Reply:
Thanks Cody, I'll change these over at some stage